File: Sobirov_Ulugbek/SmartCity/SmartCity.Core/Proxy/SecurityProxy.cs
namespace SmartCity.Core.Proxy;
public class SecurityProxy : ISecurityAccess
{
    private readonly SecuritySystem _realSystem;
    private readonly List<string> _authorizedUsers = new() { "admin", "security" };
    
    public SecurityProxy()
    {
        _realSystem = new SecuritySystem();
    }
    
    private bool CheckAccess(string user)
    {
        if (_authorizedUsers.Contains(user.ToLower()))
        {
            return true;
        }
        
        Console.WriteLine($"❌ RUXSAT RAD ETILDI: [{user}] xavfsizlik tizimiga kira olmaydi!");
        return false;
    }
    
    public void AccessSecuritySystem(string user)
    {
        Console.WriteLine($"🔐 Proxy: [{user}] uchun ruxsat tekshirilmoqda...");
        if (CheckAccess(user))
        {
            _realSystem.AccessSecuritySystem(user);
        }
    }
    
    public void ViewCameraFeed(string cameraId, string user)
    {
        Console.WriteLine($"🔐 Proxy: [{user}] uchun kamera ruxsati tekshirilmoqda...");
        if (CheckAccess(user))
        {
            _realSystem.ViewCameraFeed(cameraId, user);
        }
    }
    
    public void ControlAlarm(string action, string user)
    {
        Console.WriteLine($"🔐 Proxy: [{user}] uchun signalizatsiya ruxsati tekshirilmoqda...");
        if (CheckAccess(user))
        {
            _realSystem.ControlAlarm(action, user);
        }
    }
}

File: abdulatif.nuriddinov/core/proxy/security_proxy.py
from datetime import datetime

class SecurityProxy:
    def __init__(self, security_system):
        self._security_system = security_system

    def _log_access(self, user, action):
        log_entry = f"{datetime.now()}: User '{user}' performed '{action}'"
        print(f"[SECURITY LOG] {log_entry}")

    def get_security_status(self, user):
        self._log_access(user, "get_security_status")
        return self._security_system.get_security_status()

    def monitor_area(self, user, camera_id):
        self._log_access(user, f"monitor_area({camera_id})")
        return self._security_system.monitor_area(camera_id)
